src/main/java/org/springframework/data/elasticsearch/repository/support/ElasticsearchCollectionValueToStringConverter.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.repository.support;
+
+import org.springframework.core.convert.ConversionService;
+import org.springframework.core.convert.TypeDescriptor;
+import org.springframework.core.convert.converter.GenericConverter;
+import org.springframework.lang.Nullable;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.StringJoiner;
+
+/**
+ * Convert a collection into string for value part of the elasticsearch query.
+ * <p>
+ * The string should be wrapped with square brackets, with each element quoted therefore
```

Comment:
quoted elements means Strings? What about other types, i.e. numeric ones?

Reply:
The non string types will be converted correctly by other converters in `ConversionSerivce`. Take integers as the example:
```json
				{
				  "bool":{
				    "must":{
				      "terms":{
				        "name": [1, 2, 3]
				      }
				    }
				  }
				}
```
I'll add more tests to prove this.

Reply:
And the comment is also changed to avoid the misleading.